py/dml/compat.py

```diff
@@ -198,6 +198,30 @@ class suppress_WLOGMIXUP(CompatFeature):
     short = "Suppress the warning 'WLOGMIXUP' by default"
     last_api_version = api_6
 
+@feature
+class suppress_WBIGUNROLL(CompatFeature):
+    '''This compatibility feature makes it so the warning `WBIGUNROLL` is
+    suppressed by default. `WBIGUNROLL` warns about `#foreach` and `#select`
+    statements that compile to large unrolled loops &mdash; for more
+    information, see the documentation of `WBIGUNROLL` in the
+    [Messages](messages.html) section.
+
+    `WBIGUNROLL` is suppressed by default below Simics API version 8 in order
+    to avoid overwhelming users with warnings. Addressing occurences of large
+    unrolled loops should be done before or as part of migration to Simics API
+    version 8.
+
+    Passing `--no-compat=suppress_WBUGUNROLL` to DMLC has almost the same effect
+    as passing `--warn=WBIGUNROLL`; either will cause DMLC to report the warning
+    even when the Simics API version in use is below 8. The only difference
+    between these two options is that if `--no-compat=suppress_WBIGUNROLL` is
+    used (and `--warn=WBIGUNROLL` is not), then `WBIGUNROLL` may still be
+    explicitly suppressed via `--no-warn=WBIGUNROLL`. In contrast,
+    `--warn=WBIGUNROLL` doesn't allow for `WBIGUNROLL` to be suppressed at
+    all.'''
```

Comment:
that's somewhat weird, when you pass `--X --no-X` to a compiler, you expect that the one that appears last takes precedence.

In any case, I think we can drop this paragraph entirely; there is no essential difference and the main reason for a compat feature is for consistency with other things that change with API 8.

Reply:
I agree with dropping the explanation of the difference; that was inherited from the description of `suppress_WLOGMIXUP`, but it doesn't bear repeating. Though, I think I should keep the mention that you can either use `--warn=WBIGUNROLL` or `--no-compat=suppress_WBIGUNROLL`.

Reply:
>  I should keep the mention that you can either use --warn=WBIGUNROLL or --no-compat=suppress_WBIGUNROLL.

Agreed